fs/localfs/local_fs.go

```diff
@@ -141,6 +141,20 @@ func toDirEntryOrNil(dirEntry os.DirEntry, prefix string) (fs.Entry, error) {
 	return entryFromDirEntry(fi, prefix), nil
 }
 
+func (fsd *filesystemDirectory) IterateEntries(ctx context.Context, cb func(context.Context, fs.Entry) error) error {
+	entries, err := fsd.Readdir(ctx)
```

Comment:
all those implementations are the same, can you introduce a helper and call everywhere?